Frontend/src/ui/User.jsx

```diff
@@ -0,0 +1,53 @@
+import { useQuery } from '@tanstack/react-query'
+import React, { useContext } from 'react'
+import { getPermissions } from '../api/auth/get_permissions'
+import { CompetitionContext } from '../api/helper/context/competition_context'
+import { PermissionsContext } from '../api/helper/context/permission_context'
+import { UserContext } from '../api/helper/context/user_context'
+import getMe from '../api/user/get/get_me'
+import { setMessage } from './events/messages'
+import LoadingMessage from './messages/loadingMessage'
+
+export const USER_KEY = 'user'
+export const JWT_KEY = 'jwt'
+export default function User({ children }) {
+  const { isLoading, data: user } = useQuery({
+    queryKey: ['user-me'],
+    queryFn: () => getMe(),
+    retry: false,
+    onError: (_) => {
+      // We are not logged in, set user explicitly to undefined?
+    },
+  })
+  const { isLoading: permissionsLoading, data: permissions } = useQuery({
+    queryKey: ['permissions-me'],
+    queryFn: () => getPermissions(),
+    retry: false,
+    onError: (err) => {
+      setMessage(err.message, 'error')
+    },
+    // Don't try to get permissions if we are not logged in
+    enabled: user.id,
```

Comment:
I believe that when this query is disabled, `permissionsLoading` will be stuck as `true` (even though it's not fetching anything), which might affect the return below?

Reply:
fixed it by doing 
```
return loggedIn && isLoading ? (
    <LoadingMessage />
  )
```
in the new context provider